tests/components/proximity/test_init.py

```diff
@@ -1,557 +1,589 @@
+        "proximity": {
+            "home": {
+                "ignored_zones": ["work"],
+                "devices": ["device_tracker.test1"],
+                "tolerance": "1",
+            }
+        }
+    }
+
+    assert await async_setup_component(hass, DOMAIN, config)
+
+    hass.states.async_set(
+        "device_tracker.test1",
+        "not_home",
+        {"friendly_name": "test1", "latitude": 40.1, "longitude": 20.1},
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("proximity.home")
+    assert state.attributes.get("nearest") == "test1"
+    assert state.attributes.get("dir_of_travel") == "unknown"
+
+    hass.states.async_set(
+        "device_tracker.test1",
+        "not_home",
+        {"friendly_name": "test1", "latitude": 20.1, "longitude": 10.1},
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("proximity.home")
+    assert state.attributes.get("nearest") == "test1"
+    assert state.attributes.get("dir_of_travel") == "towards"
+
+
+async def test_all_device_trackers_in_ignored_zone(hass):
+    """Test for tracker in ignored zone."""
+    config = {
+        "proximity": {
+            "home": {
+                "ignored_zones": ["work"],
+                "devices": ["device_tracker.test1"],
+                "tolerance": "1",
+            }
+        }
+    }
+
+    assert await async_setup_component(hass, DOMAIN, config)
+
+    hass.states.async_set("device_tracker.test1", "work", {"friendly_name": "test1"})
+    await hass.async_block_till_done()
+    state = hass.states.get("proximity.home")
+    assert state.state == "not set"
+    assert state.attributes.get("nearest") == "not set"
+    assert state.attributes.get("dir_of_travel") == "not set"
+
+
+async def test_device_tracker_test1_no_coordinates(hass):
+    """Test for tracker with no coordinates."""
+    config = {
+        "proximity": {
+            "home": {
+                "ignored_zones": ["work"],
+                "devices": ["device_tracker.test1"],
+                "tolerance": "1",
+            }
+        }
+    }
+
+    assert await async_setup_component(hass, DOMAIN, config)
+
+    hass.states.async_set(
+        "device_tracker.test1", "not_home", {"friendly_name": "test1"}
+    )
+    await hass.async_block_till_done()
+    state = hass.states.get("proximity.home")
+    assert state.attributes.get("nearest") == "not set"
+    assert state.attributes.get("dir_of_travel") == "not set"
+
+
+async def test_device_tracker_test1_awayfurther_than_test2_first_test1(hass):
+    """Test for tracker ordering."""
+    config_zones(hass)
+    await hass.async_block_till_done()
+
+    hass.states.async_set(
+        "device_tracker.test1", "not_home", {"friendly_name": "test1"}
+    )
+    await hass.async_block_till_done()
+    hass.states.async_set(
+        "device_tracker.test2", "not_home", {"friendly_name": "test2"}
+    )
+    await hass.async_block_till_done()
+
+    assert await async_setup_component(
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)